### PR TITLE
Batch:SortFrontToBack2Pass(...)

### PR DESCRIPTION
**Pull request by TrevorCash**
Modify Batch::SortFrontToBack2Pass(..) to accept Batch** iterators instead of entire array.

### CODE REVIEW

File: Source/Urho3D/Graphics/Batch.cpp [hunk 1]
```diff
@@ -781,24 +781,25 @@ void BatchQueue::SortFrontToBack()
     for (auto i = batchGroups_.begin(); i != batchGroups_.end(); ++i)
         sortedBatchGroups_[index++] = &i->second;
 
-    SortFrontToBack2Pass(reinterpret_cast<ea::vector<Batch*>& >(sortedBatchGroups_));
+
+    SortFrontToBack2Pass(reinterpret_cast<Batch**>(sortedBatchGroups_.begin()), reinterpret_cast<Batch**>(sortedBatchGroups_.end()) );
```
Comment (rokups) [inline Source/Urho3D/Graphics/Batch.cpp]: This should be `static_cast<>`. Not that `reinterpret_cast<>` is wrong or wont work. Using right cast is more kosher :]

Comment (TrevorCash) [reply Source/Urho3D/Graphics/Batch.cpp]: I tried changing it to static_cast, I get the error:

'static_cast': cannot convert from 'Urho3D::BatchGroup **' to 'Urho3D::Batch **'	Urho3D	C:\Users\casht\Repos\techgame\rbfx\Source\Urho3D\Graphics\Batch.cpp	785

Comment (TrevorCash) [reply Source/Urho3D/Graphics/Batch.cpp]: Looks like it is due to it casting a double pointer and not a single pointer.  
https://stackoverflow.com/questions/26277391/why-cant-static-cast-a-double-void-pointer

**Comment by rokups** [comment]
Alright, screw it then. C++ sucks :D

**Pull request merged**